.github/workflows/blackduck_scan.yml

```diff
@@ -55,9 +50,13 @@ jobs:
           curl https://sig-repo.synopsys.com/artifactory/bds-integrations-release/com/synopsys/integration/synopsys-detect/${{ vars.SYNOPSYS_DETECT_VERSION }}/synopsys-detect-${{ vars.SYNOPSYS_DETECT_VERSION }}.jar -o $DETECT_FILE
           mkdir appsec || true
           mv $DETECT_FILE appsec/
+          BRANCH_NAME="${GITHUB_REF#refs/heads/}"
 
-          if [[ "${{ inputs.production }}" == "true" ]]; then  export RELEASE_TYPE="PROD"; export VERSION_ID="Release_${GITHUB_REF#refs/heads/}"; fi
-          if [[ "${{ inputs.production }}" == "false" ]]; then export RELEASE_TYPE="DEV"; export VERSION_ID="Development-${{ inputs.version }}"; fi
+          if [[ "${{ inputs.target }}" == "Production" && $BRANCH_NAME != "v6" && $BRANCH_NAME != "v7" ]]; \
+          then echo "Only v6 & v7 branch can be published as production"; exit 1; \
+          fi
+          if [[ "${{ inputs.target }}" == "Production" ]]; then export RELEASE_TYPE="PROD"; export VERSION_ID="Release_$BRANCH_NAME"; fi
+          if [[ "${{ inputs.target }}" != "Production" ]]; then export RELEASE_TYPE="DEV"; export VERSION_ID="${{ inputs.target }}"; fi
```

Comment:
We can reduce the two if conditions to one, by declare the default variables for development and then override them to production in the if condition.
```bash
RELEASE_TYPE="DEV"
VERSION_ID="${{ inputs.target }}"

[[ "${{ inputs.target }}" == "Production" ]] && RELEASE_TYPE="PROD" && VERSION_ID="Release_$BRANCH_NAME";
```
Note. export might be removed if it use for CLI only

Reply:
updated